File: Assets/Scripts/spawnGameObjectAtRandomLocation.js
#pragma strict

public var gameObjectToSpawn : GameObject;
public var arrayOfPoints : GameObject[];

function Start () {
	var spawnPointLocation = arrayOfPoints[Random.Range(0,arrayOfPoints.length - 1)].transform.position;
	Instantiate(gameObjectToSpawn, spawnPointLocation, Quaternion.identity);
}

File: Assets/Scripts/spawnGameObjectAtRandomLocation.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public class spawnGameObjectAtRandomLocation : MonoBehaviour
{
    public GameObject gameObjectToSpawn;
    public GameObject[] arrayOfPoints;
    public virtual void Start()
    {
        Vector3 spawnPointLocation = this.arrayOfPoints[Random.Range(0, this.arrayOfPoints.Length - 1)].transform.position;
        UnityEngine.Object.Instantiate(this.gameObjectToSpawn, spawnPointLocation, Quaternion.identity);
    }

}